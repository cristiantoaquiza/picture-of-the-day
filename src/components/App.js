import React, { useEffect, useState } from "react";

const App = () => {
  const [title, setTitle] = useState();
  const [explanation, setExplanation] = useState();
  const [url, setUrl] = useState();
  const [date, setDate] = useState();

  const fetchData = async () => {
    const currentDate = new Date();
    const year = currentDate.getFullYear();
    const month = currentDate.getMonth() + 1;
    const day = currentDate.getDate();
    const response = await fetch(
      `https://api.nasa.gov/planetary/apod?api_key=DEMO_KEY&date=${year}-${month}-${day}`
    );
    const data = await response.json();
    const { title, explanation, url, date } = data;
    setTitle(title);
    setExplanation(explanation);
    setUrl(url);
    setDate(date);
  };

  useEffect(() => {
    fetchData();
  }, []);

  return (
    <div>
      <h1>Astronomy Picture of the Day</h1>
      <h2>{title}</h2>
      <img src={url} />
      <p>{explanation}</p>
      <span>{date}</span>
    </div>
  );
};

export default App;
