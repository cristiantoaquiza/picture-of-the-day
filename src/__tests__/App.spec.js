import React from 'react'
import renderer from 'react-test-renderer'
import App from '../App'

describe('App', () => {
  it('should test something', () => {
    expect(false).toBeFalsy()
  })

  it('should match snapshot rendered', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})