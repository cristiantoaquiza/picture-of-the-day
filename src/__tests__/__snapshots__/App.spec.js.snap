// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot rendered 1`] = `
<div>
  <h1>
    Hello from React Component
  </h1>
</div>
`;
